subprojects/core/src/main/java/org/gradle/api/internal/FeaturePreviews.java

```diff
@@ -17,18 +17,39 @@
 
 import org.gradle.StartParameter;
 
+import java.util.EnumSet;
+
 public class FeaturePreviews {
-    private StartParameter startParameter;
+    private enum Feature {
+        ADVANCED_POM_SUPPORT,
+        GRADLE_METADATA
+    }
+
+    private EnumSet<Feature> enabledFeatures = EnumSet.noneOf(Feature.class);
 
     public FeaturePreviews(StartParameter startParameter) {
-        this.startParameter = startParameter;
+        if (startParameter.isAdvancedPomSupport()) {
+            enabledFeatures.add(Feature.ADVANCED_POM_SUPPORT);
+        }
+        if (startParameter.isGradleMetadata()) {
+            enabledFeatures.add(Feature.GRADLE_METADATA);
+        }
     }
 
-    public boolean isAdvancedPomSupportEnabled() {
-        return startParameter.isAdvancedPomSupport();
+    public void enableFeature(String name) {
+        try {
+            enabledFeatures.add(Feature.valueOf(name));
+        } catch (IllegalArgumentException e) {
+            // Ignore unknown features
+        }
     }
 
-    public boolean isGradleMetadataEnabled() {
-        return startParameter.isGradleMetadata();
+    public boolean isFeatureEnabled(String name) {
```

Comment:
Internally, it probably makes sense for us to use the Enum values directly. That way I can easily find usages of `ADVANCED_POM_SUPPORT`. We just don't want these Enum values to end up as part of our public API.